### PR TITLE
Add support and tests for otj-conservedheaders-reactive

### PR DESCRIPTION
**Pull request by scottjohnson**
Depends on https://github.com/opentable/otj-conservedheaders/pull/12, which should be merged and otj-conservedheaders-reactive dep should be released and firmed down in this PR.

### CODE REVIEW

File: otj-server-reactive/pom.xml [hunk 1]
```diff
@@ -26,10 +26,6 @@
     <artifactId>otj-server-reactive</artifactId>
     <description>Provides Spring WebFlux wiring into otj-stack</description>
 
-    <properties>
```
Comment (mikebell90) [inline otj-server-reactive/pom.xml]: That was my favorite part of the PR!

File: otj-server-reactive/src/main/java/com/opentable/server/reactive/ReactiveServerCommonConfiguration.java [hunk 2]
```diff
@@ -40,7 +41,10 @@
         EmbeddedJettyConfiguration.class,
         EmbeddedReactiveJetty.class,
         // Log server request information
+        // TODO: Remove this after updating servers that use ServerRequestLog directly
```
Comment (mikebell90) [inline otj-server-reactive/src/main/java/com/opentable/server/reactive/ReactiveServerCommonConfiguration.java]: Explain why in your comment

File: pom.xml [hunk 3]
```diff
@@ -47,6 +47,9 @@
 
         <!-- Remove when parent is updated -->
         <dep.otj-logging.version>3.0.2</dep.otj-logging.version>
+
+        <!-- Remove when otj-parent is updated -->
```
Comment (mikebell90) [inline pom.xml]: No, more accurately when the other is cut. We don't publish snapshots

File: pom.xml [hunk 4]
```diff
@@ -47,6 +47,9 @@
 
         <!-- Remove when parent is updated -->
         <dep.otj-logging.version>3.0.2</dep.otj-logging.version>
```
Comment (mikebell90) [inline pom.xml]: Parent is updated, please remove.

File: otj-server-reactive/src/main/java/com/opentable/server/reactive/ReactiveServerCommonConfiguration.java (hunk 2 above)
Comment (mikebell90) [reply otj-server-reactive/src/main/java/com/opentable/server/reactive/ReactiveServerCommonConfiguration.java]: n/m, you do below

File: otj-server-reactive/src/test/java/com/opentable/server/reactive/TestReactiveServerConfiguration.java [hunk 5]
```diff
@@ -55,15 +54,32 @@ WebClient webClient() {
...
 
+    /**
+     * This scheduler is used by {@link ConservedHeadersWebFilterTest}
+     * to set up fewer available threads than the number of requests we want to make in parallel, to ensure that
+     * reactive operations from different requests are scheduled on threads that were previously used by another
+     * request.
+     */
+    @Bean
+    Scheduler getScheduler() {
+        return Schedulers.newParallel("parallel-scheduler", 3);
```
Comment (mikebell90) [inline otj-server-reactive/src/test/java/com/opentable/server/reactive/TestReactiveServerConfiguration.java]: ignorant: any reason you chose this scheduler as oppoosed to the gajillion others (the size I understand)

File: otj-server-reactive/src/test/java/com/opentable/server/reactive/TestReactiveServerConfiguration.java [hunk 6]
```diff
@@ -55,15 +54,32 @@ WebClient webClient() {
...
         private WebClient webClient;
 
+        @Autowired
+        private Scheduler parallelScheduler;
+
         @RequestMapping("test")
         public Mono<String> test() {
+            LOG.info("Called 'test' endpoint");
             return Mono.just("test");
         }
```
Comment (mikebell90) [inline otj-server-reactive/src/test/java/com/opentable/server/reactive/TestReactiveServerConfiguration.java]: btw I just noticed echo - isn't that an outside service this test depends on.

File: otj-server-reactive/src/main/java/com/opentable/server/reactive/logging/ReactiveServerLoggingConfiguration.java [hunk 7]
```diff
@@ -24,6 +24,7 @@
 /**
  * Configures automatic server request logging using Jetty for reactive applications.
  */
```
Comment (adamzr) [inline otj-server-reactive/src/main/java/com/opentable/server/reactive/logging/ReactiveServerLoggingConfiguration.java]: Add an `@deprecated` segment to the JavaDoc, so people know what is supposed to be done instead

File: otj-server-reactive/src/test/java/com/opentable/server/reactive/ConservedHeadersWebFilterTest.java [hunk 8]
```diff
@@ -0,0 +1,223 @@
...
+ * - If no request-id header is included, it is added in automatically.
+ * - Log messages associated with the request contain the appropriate conserved header values.
+ * - When reactive operations move between threads (see {@link TestReactiveServerConfiguration} /api/threading)
+ * conserved header information continues to be associated with the request.
+ * - When multiple requests are in-flight simultaneously, conserved header information is associated with the
+ * correct request, even when reactive operations are continued on a thread previously used by a different request.
+ */
+public class ConservedHeadersWebFilterTest extends AbstractTest {
+
+    private static final org.slf4j.Logger LOG = LoggerFactory.getLogger(ConservedHeadersWebFilterTest.class);
```
Comment (adamzr) [inline otj-server-reactive/src/test/java/com/opentable/server/reactive/ConservedHeadersWebFilterTest.java]: Import `org.slf4j.Logger`

File: pom.xml (hunk 3 above)
Comment (scottjohnson) [reply pom.xml]: Yes _and_ also after the other is cut we will still need to update this once the parent is updated. Was just thinking ahead.

File: otj-server-reactive/src/test/java/com/opentable/server/reactive/TestReactiveServerConfiguration.java (hunk 5 above)
Comment (scottjohnson) [reply otj-server-reactive/src/test/java/com/opentable/server/reactive/TestReactiveServerConfiguration.java]: Relatively arbitrary. It creates a new fixed-size scheduler that drops in to the Reactor scheduler interface.

File: otj-server-reactive/src/test/java/com/opentable/server/reactive/TestReactiveServerConfiguration.java (hunk 6 above)
Comment (scottjohnson) [reply otj-server-reactive/src/test/java/com/opentable/server/reactive/TestReactiveServerConfiguration.java]: As discussed, we need to be able to call out to some sort of external service. This is already done in otj-server-mvc. We could consider refactoring and testing in a different way (WireMock/MockServer?) for both modules.

File: otj-server-reactive/src/main/java/com/opentable/server/reactive/logging/ReactiveServerLoggingConfiguration.java (hunk 7 above)
Comment (scottjohnson) [reply otj-server-reactive/src/main/java/com/opentable/server/reactive/logging/ReactiveServerLoggingConfiguration.java]: 👍

File: otj-server-reactive/src/test/java/com/opentable/server/reactive/ConservedHeadersWebFilterTest.java (hunk 8 above)
Comment (scottjohnson) [reply otj-server-reactive/src/test/java/com/opentable/server/reactive/ConservedHeadersWebFilterTest.java]: 👍

**Pull request merged**